lib/alertmanager_api/silence.py

```diff
@@ -0,0 +1,198 @@
+    """
+    class implementing an interface to the Kayobe's alertmanager
+    functionalities:
+        * create a silence
+        * remove an existing silence
+    """
+
+    def __init__(self, alertmanager_account):
+        """
+        :param alertmanager_account: object with credentials
+        :type alertmanager_account: AlertManagerAccount
+        """
+        self.log = logging.getLogger("AlertManagerAPI")
+        self.alertmanager_account = alertmanager_account
+        self.auth = HTTPBasicAuth(
+            alertmanager_account.username, alertmanager_account.password
+        )
+        self.api_endpoint = alertmanager_account.alertmanager_endpoint
+
+    def schedule_silence(self, silence):
+        """
+        Schedules a silence in alertmanager
+
+        :param silence: object with the specs to create a silence
+        :type silence: SilenceDetails
+        :return: a new silence object, including the silence_id
+        :rtype: SilenceDetails
+        :raises MissingMandatoryParamError:
+            if any of the required attributes are missing
+            in the silence object:
+            instance_name
+            start time
+            end time
+            author
+            comment
+        :raises requests.exceptions.RequestException:
+            when the request to the AlertManager failed
+        """
+        if not silence.instance_name:
+            msg = "Missing silence instance name"
+            self.log.critical(msg)
+            raise MissingMandatoryParamError(msg)
+        if not silence.start_time_dt:
+            msg = "Missing silence start time"
+            self.log.critical(msg)
+            raise MissingMandatoryParamError(msg)
+        if not silence.end_time_dt:
+            msg = "Missing silence end time"
+            self.log.critical(msg)
+            raise MissingMandatoryParamError(msg)
+        if not silence.author:
+            msg = "Missing silence author"
+            self.log.critical(msg)
+            raise MissingMandatoryParamError(msg)
+        if not silence.comment:
+            msg = "Missing silence comment"
+            self.log.critical(msg)
+            raise MissingMandatoryParamError(msg)
+
+        payload = {
+            "matchers": [
+                {
+                    "name": "instance",
+                    "value": silence.instance_name,
+                    "isRegex": False,
+                }
+            ],
+            "startsAt": silence.start_time_str,
+            "endsAt": silence.end_time_str,
+            "createdBy": silence.author,
+            "comment": silence.comment,
+        }
+        try:
+            api_url = f"{self.api_endpoint}/api/v2/silences"
+            response = requests.post(
+                api_url,
+                auth=self.auth,
+                headers={"Accept": "application/json"},
+                json=payload,
+                timeout=10,
+            )
+            response.raise_for_status()
```

Comment:
only this part should be in the try-catch block - limit the try block to only those calls that we expect to error to occur in - its just good practice